stacks-signer/src/runloop.rs

```diff
@@ -0,0 +1,439 @@
+                    &self.signing_round.public_keys,
+                    coordinator_public_key,
+                ) {
+                    Some(message)
+                } else {
+                    None
+                }
+            })
+            .collect();
+        // First process all messages as a signer
+        let mut outbound_messages =
+            process_inbound_messages(&mut self.signing_round, inbound_messages.clone())
+                .unwrap_or_default();
+        // If the signer is the coordinator, then next process the message as the coordinator
+        let (messages, results) = if self.signing_round.signer.signer_id == coordinator_id {
+            self.coordinator
+                .process_inbound_messages(inbound_messages)
+                .unwrap_or_default()
+        } else {
+            (vec![], vec![])
+        };
+        outbound_messages.extend(messages);
+        (outbound_messages, results)
+    }
+}
+
+impl RunLoop<FrostCoordinator> {
+    /// Creates new runloop from a config and command
+    pub fn new(config: &Config, command: RunLoopCommand) -> Self {
+        // TODO: should this be a config option?
+        // The threshold is 70% of the shares
+        let threshold = (config.signer_ids_public_keys.key_ids.len() / 10 * 7)
+            .try_into()
+            .unwrap();
+        let total_signers = config
+            .signer_ids_public_keys
+            .signers
+            .len()
+            .try_into()
+            .unwrap();
+        let total_keys = config
+            .signer_ids_public_keys
+            .key_ids
+            .len()
+            .try_into()
+            .unwrap();
+        let key_ids = config
+            .signer_key_ids
+            .get(&config.signer_id)
+            .unwrap()
+            .iter()
+            .map(|i| i - 1) // SigningRound::new (unlike SigningRound::from) doesn't do this
+            .collect::<Vec<u32>>();
+        RunLoop {
+            event_timeout: config.event_timeout,
+            coordinator: FrostCoordinator::new(total_signers, total_keys, threshold, config.message_private_key),
+            signing_round: SigningRound::new(
+                threshold,
+                total_signers,
+                total_keys,
+                config.signer_id,
+                key_ids,
+                config.message_private_key,
+                config.signer_ids_public_keys.clone(),
+            ),
+            stacks_client: StacksClient::from(config),
+            command,
+            state: State::Idle,
+        }
+    }
+}
+
+/// Process inbound messages using the frost_signer signing round mechanism
+pub fn process_inbound_messages(
+    signing_round: &mut SigningRound,
+    messages: Vec<Message>,
+) -> Result<Vec<Message>, frost_signer::signing_round::Error> {
+    let mut responses = vec![];
+    for message in messages {
+        // TODO: this code was swiped from frost-signer. Expose it there so we don't have duplicate code
```

Comment:
Flagging this TODO for completion before merge.

Reply:
`frost-signer` is part of `stacks-sbtc`, so IMO these TODOs should be part of follow on PR after we've made a number of changes to `wsts` and `stacks-sbtc`.

Reply:
Gotcha.  Just trying to make sure that they don't get dropped on the floor :pray: